packages/hardhat-core/src/builtin-tasks/flatten.ts

```diff
@@ -56,35 +61,64 @@ function getFileWithoutImports(resolvedFile: ResolvedFile) {
 
 subtask(
   TASK_FLATTEN_GET_FLATTENED_SOURCE,
-  "Returns all contracts and their dependencies flattened",
-  async (_, { run }) => {
+  "Returns all contracts and their dependencies flattened"
+)
+  .addOptionalParam("files", undefined, undefined, types.any)
+  .setAction(async ({ files }: { files?: string[] }, { run }) => {
+    const dependencyGraph: DependencyGraph = await run(
+      TASK_FLATTEN_GET_DEPENDENCY_GRAPH,
+      { files }
+    );
+
     let flattened = "";
 
-    const graph: DependencyGraph = await run(
-      TASK_COMPILE_SOLIDITY_GET_DEPENDENCY_GRAPH
-    );
-    if (graph.getResolvedFiles().length === 0) {
+    if (dependencyGraph.getResolvedFiles().length === 0) {
       return flattened;
     }
 
     const packageJson = await getPackageJson();
     flattened += `// Sources flattened with hardhat v${packageJson.version} https://usehardhat.com`;
 
-    const sortedFiles = getSortedFiles(graph);
+    const sortedFiles = getSortedFiles(dependencyGraph);
 
     for (const file of sortedFiles) {
       flattened += `\n\n// File ${file.getVersionedName()}\n`;
       flattened += `\n${getFileWithoutImports(file)}\n`;
     }
 
     return flattened.trim();
-  }
-);
+  });
+
+subtask(TASK_FLATTEN_GET_DEPENDENCY_GRAPH)
+  .addOptionalParam("files", undefined, undefined, types.any)
+  .setAction(async ({ files }: { files: string[] | undefined }, { run }) => {
+    const sourcePaths =
```

Comment:
```suggestion
    const sourcePaths: string[] =
```

Reply:
Oh, this is really interesting. We should probably always declare the types of the variables that hold values returned by `run`, right?